src/net/socket.rs

```diff
@@ -49,6 +51,13 @@ impl Socket {
         ))
     }
 
+    /// Configure the socket with the passed configuration.
+    pub fn with_config(mut self, config: Config) -> Socket {
```

Comment:
I think I would prefer to have a `bind_with_config` method which is similar to the bind method today, as well as a `bind` method which calls it with `Config::default()`